base/abstractarray.jl

```diff
@@ -2299,8 +2302,9 @@ function _typed_hvncat(::Type{T}, shape::Tuple{Vararg{Tuple, N}}, row_first::Boo
     shapepos = ones(Int, nd)
 
     for i ∈ eachindex(as)
+        length(as[i]) > 0 || throw(ArgumentError("argument $i has no elements"))
```

Comment:
I don't see why this is a special case, i.e. why isn't it covered by the other check that the number of elements matches?

Reply:
For example `[zeros(1,2,0);;;7 8]` doesn't need to be an error.

Reply:
The former check is the length of as; this is the length of each element. Though I should probably be using cat_length here in case any of them are strings.

But you're right. This is a crude check. The dimension calculating algorithm right now just can't handle arrays with a zero dimension. e.g. [zeros(1, 0);;;7 8] creates an underfilled array. Seems better to make them error for now until I can address that.